Makefile

```diff
@@ -221,8 +267,16 @@ sdl: ${SDL} libhl
 openal: ${OPENAL} libhl
 	${CC} ${CFLAGS} -shared -o openal.hdll ${OPENAL} ${LIBFLAGS} -L. -lhl $(LIBOPENAL)
 
+./include/mbedtls/%.o: ./include/mbedtls/%.c
+	${CC} ${CFLAGS} -o $@ -c $< ${SSL_CFLAGS}
+
+# force rebuild ssl.o in case we mix SSL_STATIC with normal build
+.PHONY: libs/ssl/ssl.o
+libs/ssl/ssl.o: libs/ssl/ssl.c
+	${CC} ${CFLAGS} -o $@ -c $< ${SSL_CFLAGS}
```

Comment:
Would it work if the `libs/ssl/ssl.c` dependency is removed? like this:
```makefile
libs/ssl/ssl.o:
	${CC} ${CFLAGS} -o $@ -c libs/ssl/ssl.c ${SSL_CFLAGS}
```

.PHONY implies that `libs/ssl/ssl.o` doesn't exist in the file system: https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

Reply:
The rule `libs/ssl/ssl.o` will not be executed if a file of the same name already exists in the filesystem, and the prerequisite has not changed after the modified time of the `.o`.
.PHONY force it to ignore the file.

Reply:
So removing the dependency will not force the rebuild, as ssl.o already exists;
it's possible to put `rm -f libs/ssl/ssl.o` somewhere to force the rebuild, but I didn't find a good place yet

Reply:
> The rule libs/ssl/ssl.o will not be executed if a file of the same name already exists in the filesystem, and all the prerequisite's modification time is before the modification time of the .o.

Makes sense! I guess `.PHONY` works for now then. A better solution might be possible if we write the value of `SSL_STATIC` to a file whenever it changes, and then add that file as a target dependency.

Reply:
Hum add a tmp file can be a good choice, I've already seen such usage in other projects, let me check

Reply:
I think use .PHONY is simpler in this case, even it's not very flexible (e.g. replace `ssl.o` to `%.o` will not work, add other c files will probably not work either). Let us stay with it for now, and we can improve it later

Reply:
Sounds good!